Branch gran dt

Ulises-Berasain opened the pull request.

Ulises-Berasain committed:
Inicio Gran DT

comienzo de la estructura y algo funcional de la pagina sobre el juego de Gran DT con alguna fallas aun.

Ulises-Berasain committed:
Cambio de formacion

agrego las demas formaciones para seleccionar

Ulises-Berasain committed:
Modulos

Separo todo en modulos ademas de agregar Cypress para que pueda hacer un auto test

fsodano commented on index.html:
```diff
@@ -0,0 +1,70 @@
+            </div>
+          </div>
+            <div class="col-8">
+                <div class="card" style="width: 20rem;" id="contenedor-cancha">
```
en gral los estilos van en CSS

fsodano commented on index.html:
```diff
@@ -0,0 +1,70 @@
+
+    <div class="container" id="contenedor">
+        <div id="contenedor-presupuesto">
+            <label class="presupuesto" id="presupuesto"><strong id="texto-presupuesto">Presupuesto</strong><p id="dinero">65000000</p></label>
```
si el texto es presupuesto, el id debería ser presupuesto en vez de dinero. Tenés que intentar tener un lenguaje ubicuo que sea igual en todos lados, porque cuando tu aplicación crece es difícil acordarse que "dinero" y "presupuesto" en realidad apuntan a la misma cosa.

Ulises-Berasain replied:
Claro, se me paso de largo eso por no pensar bien claro, le voy a colocar de id "cantidad-presupuesto" así sigue todo lo mismo para que no se pise con la otra id "presupuesto"

fsodano commented on package.json:
```diff
@@ -0,0 +1,29 @@
+  },
+  "homepage": "https://github.com/Ulises-Berasain/gran-dt#readme",
+  "dependencies": {
+    "http-server": "^0.12.3",
```
esto es una dev dependency

fsodano commented on package.json:
```diff
@@ -0,0 +1,29 @@
+  "homepage": "https://github.com/Ulises-Berasain/gran-dt#readme",
+  "dependencies": {
+    "http-server": "^0.12.3",
+    "jquery": "^3.5.1"
```
jQuery lo usás para cypress? en ese caso es una dev dependency.

fsodano commented on src/api/api.js:
```diff
@@ -0,0 +1,10 @@
+        .then(r => r.json())
+        .then(r => {
+            const {count: totalJugadores, results: jugadores} = r;
+            mostrarListaJugadores(jugadores);
```
tu API no debería saber de la UI.

Ulises-Berasain replied:
Ahí voy a tratar que con iniciar devuelva los jugadores así la function de mostrarListaJugadores la pongo en el main

fsodano commented on src/servicio/formaciones.js:
```diff
@@ -0,0 +1,217 @@
+
+const $cancha = document.createElement("img");
+const $arquero = document.createElement("img");
+const $jugador1 = document.createElement("img");
+const $jugador2 = document.createElement("img");
+const $jugador3 = document.createElement("img");
+const $jugador4 = document.createElement("img");
+const $jugador5 = document.createElement("img");
+const $jugador6 = document.createElement("img");
+const $jugador7 = document.createElement("img");
+const $jugador8 = document.createElement("img");
+const $jugador9 = document.createElement("img");
+const $jugador10 = document.createElement("img");
```
esto va en un for, y lo metés en el array `jugadores`.

Ulises-Berasain committed:
Correcciones

Cambio la ID "dinero" por algo mas entendible y especifico, modifico la api para no recibir nada del UI, cambio las dependencias y pasos los estilos que habian en el html en el css.

Ulises-Berasain committed:
Correccion

Cambio los "/src" por "../" para mejor entendimiento

Ulises-Berasain committed:
Nuevas interacciones

Se agrego una lista de titulares y de suplentes donde va a aparecer el nombre del jugador seleccionado, cada jugador seleccionado va a contar con un boton para que el usuario pueda eliminar algun jugador no deseado o para controlar mejor su presupuesto, se modificio la imagen de la silueta por una mejor y se agrego los nuevos cambios al testeo con Cypress

Ulises-Berasain committed:
Responsive

Esta pensado desde un celular de un width de 360px subiendo asi en 480px y luego 1400px para diferentes dispositivos
